### PR TITLE
fix player profile not never loading in chat

### PR DESCRIPTION
**Pull request by kingstefan26**
fix player profiles never loading in chat + a lil refactor

### CODE REVIEW

**Comment by kingstefan26** [comment]
Note: since now hypixel saves "last_saved" per profile not member per profile, it breaks up knowing if what profile the player is currently on. A full fix would involve implementing a profile switcher button that shows different profiles of a player

**Comment by cyoung06** [comment]
Would be great to refactor it this way
- Cache whole player profile's instead
- Add the button to change profile like you have suggested

**Comment by kingstefan26** [comment]
```
java.lang.NullPointerException at kr.syeyoung.dungeonsguide.features.impl.party.playerpreview.api.playerprofile.PlayerProfileParser.calculateLilyWeight(PlayerProfileParser.java:256)
         	at kr.syeyoung.dungeonsguide.features.impl.party.playerpreview.api.playerprofile.PlayerProfileParser.parseProfile(PlayerProfileParser.java:220)
         	at kr.syeyoung.dungeonsguide.features.impl.party.playerpreview.api.ApiFetcher.fetchPlayerProfiles(ApiFetcher.java:305)
         	at kr.syeyoung.dungeonsguide.features.impl.party.playerpreview.api.ApiFetcher.lambda$fetchMostRecentProfileAsync$2(ApiFetcher.java:157)

```
Happens but it doesn't interrupt normal operation

**Comment by cyoung06** [comment]
ooh nice refactoring

File: src/main/java/kr/syeyoung/dungeonsguide/features/impl/party/playerpreview/FeatureViewPlayerStatsOnJoin.java [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+            Gui.drawRect(0, 193, 90, 220, 0xFF23272a);
+            Gui.drawRect(2, 195, 88, 218, new Rectangle(2, 195, 86, 23).contains(relX, relY) ? 0xFF859DF0 : 0xFF7289da);
+            Gui.drawRect(90, 193, 180, 220, 0xFF23272a);
+            Gui.drawRect(91, 195, 178, 218, new Rectangle(91, 195, 176, 23).contains(relX, relY) ? 0xFFdace72 : 0xdddace72);
+            GlStateManager.enableBlend();
+            GL14.glBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_ALPHA);
+            GlStateManager.tryBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_ALPHA);
+            fr.drawString("Kick", (90 - fr.getStringWidth("Kick")) / 2, (364 - fr.FONT_HEIGHT) / 2, 0xFFFFFFFF);
+            fr.drawString("Invite", (90 - fr.getStringWidth("Invite")) / 2, (414 - fr.FONT_HEIGHT) / 2, 0xFFFFFFFF);
+            fr.drawString("Change Profile", ((90 - fr.getStringWidth("Change Profile")) / 2) + 90, (414 - fr.FONT_HEIGHT) / 2, 0xFFFFFFFF);
```
Comment (cyoung06) [inline src/main/java/kr/syeyoung/dungeonsguide/features/impl/party/playerpreview/FeatureViewPlayerStatsOnJoin.java:219]: A room for improvement
- Hovering highlights button
- maybe put the button next to the view inventory button? or arrows on the side of player

**Comment by cyoung06** [comment]
Also please check that I have changed  build.gradle into arch loom

**Comment by Ascynx** [comment]
:eyes:

**Comment by kingstefan26** [comment]
> Also please check that I have changed build.gradle into arch loom

i in fact did

**Comment by kingstefan26** [comment]
OK YOU CAN MERGE ITS ALL FINE AND GOOD

**Comment by kingstefan26** [comment]
(aside the exception i mentioned above if it bothers you fix it urself (prob hypixel messing with some field in the json ))

**Review by cyoung06 — APPROVED** [review]
Ehhhhhhhhhhhhh
we should rewrite viewing player stats using the-good-old-dg-`MPanel` (TM)s
but we can do it later.

The rest looks good to me.

**Pull request merged**